src/Tizen.Uix.VoiceControlManager/Tizen.Uix.VoiceControlManager/VoiceControlManagerClient.cs

```diff
@@ -0,0 +1,2105 @@
+        {
+            get
+            {
+                RecognitionModeType mode;
+
+                ErrorCode error = VcMgrGetRecognitionMode(out mode);
+                if (error != ErrorCode.None)
+                {
+                    Log.Error(LogTag, "GetRecognitionMode Failed with error " + error);
+                    return RecognitionModeType.Undefined;
+                }
+
+                return mode;
+            }
+            set
+            {
+                ErrorCode error = VcMgrSetRecognitionMode(value);
+                if (error != ErrorCode.None)
+                {
+                    Log.Error(LogTag, "SetRecognitionMode Failed with error " + error);
+                    throw ExceptionFactory.CreateException(error);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets private data between app and engine.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>
+        /// http://tizen.org/privilege/voicecontrol.manager
+        /// </privilege>
+        /// <privlevel>
+        /// public
+        /// </privlevel>
+        /// <feature>
+        /// http://tizen.org/feature/speech.control
+        /// http://tizen.org/feature/microphone
+        /// </feature>
+        /// <param name="key">Private key</param>
+        /// <param name="data">Private data</param>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="InvalidOperationException">This exception can be due to operation failed.</exception>
+        /// <exception cref="OutOfMemoryException">This exception can be due to out of memory.</exception>
+        /// <pre>
+        /// The State must be Ready.
+        /// </pre>
+        public static void SetPrivateData(string key, string data)
+        {
+            ErrorCode error = VcMgrSetPrivateData(key, data);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "SetPrivateData Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// Gets private data between app and engine.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <privilege>
+        /// http://tizen.org/privilege/voicecontrol.manager
+        /// </privilege>
+        /// <privlevel>
+        /// public
+        /// </privlevel>
+        /// <feature>
+        /// http://tizen.org/feature/speech.control
+        /// http://tizen.org/feature/microphone
+        /// </feature>
+        /// <param name="key">Private key</param>
+        /// <returns>Private data</returns>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="InvalidOperationException">This exception can be due to operation failed.</exception>
```

Comment:
If this exception can be thrown in both cases, please write it as a single instance of `<exception cref="InvalidOperationException">`.